diff --git a/includes/file.inc b/includes/file.inc
index ba3da06..520786c 100644
--- a/includes/file.inc
+++ b/includes/file.inc
@@ -1820,6 +1820,27 @@ function file_validate_image_resolution(stdClass $file, $maximum_dimensions = 0,
           image_scale($image, $width, $height);
           image_save($image);
           $file->filesize = $image->info['file_size'];
+          // If we have minimum size, check immediately
+          // and don't show success-scaled message if there is a minimum size error.
+          if ($minimum_dimensions) {
+            // Get info for scaled image.
+            $info = image_get_info($image->source);
+            // Check that it is larger than the given dimensions.
+            list($width, $height) = explode('x', $minimum_dimensions);
+            if ($info['width'] < $width || $info['height'] < $height) {
+              $errors[] = t(
+                'Image was scaled down to fit %max_dimensions pixels, but as a result one of the sides less than required minimum dimensions %min_dimensions pixels.',
+                array('%max_dimensions' => $maximum_dimensions, '%min_dimensions' => $minimum_dimensions)
+              );
+            }
+            else {
+              drupal_set_message(t('The image was resized to fit within the maximum allowed dimensions of %dimensions pixels.', array('%dimensions' => $maximum_dimensions)));
+            }
+
+            // Here actually we pass both validations for minimum and maximum.
+            return $errors;
+          }
+
           drupal_set_message(t('The image was resized to fit within the maximum allowed dimensions of %dimensions pixels.', array('%dimensions' => $maximum_dimensions)));
         }
         else {
