diff --git a/single_user_role.module b/single_user_role.module
index a00617e..8d38796 100755
--- a/single_user_role.module
+++ b/single_user_role.module
@@ -53,7 +53,6 @@ function single_user_role_menu() {
     'page arguments' => array('single_user_role_config_form'),
     'access arguments' => array('single user role config'),
     'file' => 'single_user_role.admin.inc',
-    'type' => MENU_CALLBACK,
     'weight' => 10,
   );
   return $items;
