diff --git a/profile2.module b/profile2.module
index c0b89ab..d40891f 100644
--- a/profile2.module
+++ b/profile2.module
@@ -87,6 +87,7 @@ function profile2_entity_info() {
       'path' => 'admin/structure/profiles',
       'file' => 'profile2.admin.inc',
       'controller class' => 'Profile2TypeUIController',
+      'menu wildcard' => '%profile2_type',
     ),
   );
 
@@ -128,6 +129,26 @@ function profile2_menu() {
 }
 
 /**
+ * Implements hook_admin_menu_map().
+ *
+ * Provide sub item for each template bundle.
+ */
+function profile2_admin_menu_map() {
+  if (!user_access('administer profiles')) {
+    return;
+  }
+
+  $map['admin/structure/profiles/manage/%profile2_type'] = array(
+    'parent' => 'admin/structure/profiles',
+    'arguments' => array(
+      array('%profile2_type' => array_keys(profile2_get_types()))
+    ),
+  );
+
+  return $map;
+}
+
+/**
  * Implements hook_permission().
  */
 function profile2_permission() {
