diff --git a/ife.module b/ife.module
index 7720962..b8224a3 100644
--- a/ife.module
+++ b/ife.module
@@ -10,6 +10,7 @@ define('IFE_MESSAGE_ALTERNATE', 1);
 define('IFE_MESSAGE_REMOVE', 2);
 define('IFE_POSITION_INLINE_MESSAGE_BEFORE', 0);
 define('IFE_POSITION_INLINE_MESSAGE_AFTER', 1);
+define('IFE_POSITION_INLINE_MESSAGE_CUSTOM', 2);
 
 /**
  * Implements hook_menu().
@@ -260,9 +261,11 @@ function ife_element_errors_set(&$element, $display, $carry_down = FALSE) {
       // Add the position for the inline error message.
       $element['#ife_error_position'] = variable_get('ife_position_inline_message', IFE_POSITION_INLINE_MESSAGE_AFTER);
 
-      // Add in our theme wrapper on the fly.
-      if (!isset($element['#theme_wrappers']) || !in_array('ife_form_element', $element['#theme_wrappers'])) {
-        $element['#theme_wrappers'][] = 'ife_form_element';
+      // Add in our theme wrapper on the fly if it's not custom placement.
+      if ($element['#ife_error_position'] != IFE_POSITION_INLINE_MESSAGE_CUSTOM) {
+        if (!isset($element['#theme_wrappers']) || !in_array('ife_form_element', $element['#theme_wrappers'])) {
+          $element['#theme_wrappers'][] = 'ife_form_element';
+        }
       }
 
       // Found a matching error, no need to continue.
diff --git a/ife.settings.inc b/ife.settings.inc
index cee3cbc..2413fb8 100644
--- a/ife.settings.inc
+++ b/ife.settings.inc
@@ -40,11 +40,12 @@ function ife_settings_form($form, $form_state) {
   $form['general_settings']['ife_position_inline_message'] = array(
     '#type' => 'radios',
     '#title' => t('Inline error message position'),
-    '#description' => t('The position of the inline error message is relative to the element.'),
+    '#description' => t('The position of the inline error message is relative to the element. Custom type imply that developer is responsible for the output of an error in the right place.'),
     '#default_value' => variable_get('ife_position_inline_message', IFE_POSITION_INLINE_MESSAGE_AFTER),
     '#options' => array(
       IFE_POSITION_INLINE_MESSAGE_BEFORE => t('Before'),
       IFE_POSITION_INLINE_MESSAGE_AFTER => t('After'),
+      IFE_POSITION_INLINE_MESSAGE_CUSTOM => t('Custom'),
     ),
   );
 
