diff --git a/modules/file/file.field.inc b/modules/file/file.field.inc
index 59b547a..c05ab5c 100644
--- a/modules/file/file.field.inc
+++ b/modules/file/file.field.inc
@@ -183,7 +183,7 @@ function file_field_load($entity_type, $entities, $field, $instances, $langcode,
     foreach ($items[$id] as $delta => $item) {
       // If the file does not exist, mark the entire item as empty.
       if (empty($item['fid']) || !isset($files[$item['fid']])) {
-        $items[$id][$delta] = NULL;
+        unset($items[$id][$delta]);
       }
       else {
         $items[$id][$delta] = array_merge((array) $files[$item['fid']], $item);
