diff --git a/modules/user/user.module b/modules/user/user.module
index 9b00392..da7bfe4 100644
--- a/modules/user/user.module
+++ b/modules/user/user.module
@@ -2233,7 +2233,8 @@ function user_login_final_validate($form, &$form_state) {
       // exactly what the user typed in. $form_state['values']['name'] may have
       // been modified by validation handlers that ran earlier than this one.
       $query = isset($form_state['input']['name']) ? array('name' => $form_state['input']['name']) : array();
-      form_set_error('name', t('Sorry, unrecognized username or password. <a href="@password">Have you forgotten your password?</a>', array('@password' => url('user/password', array('query' => $query)))));
+      form_set_error('name');
+      form_set_error('pass', t('Sorry, unrecognized username or password. <a href="@password">Have you forgotten your password?</a>', array('@password' => url('user/password', array('query' => $query)))));
       watchdog('user', 'Login attempt failed for %user.', array('%user' => $form_state['values']['name']));
     }
   }
