diff --git a/contrib/profile2_page.inc b/contrib/profile2_page.inc
index cc3ba41..315cf5f 100644
--- a/contrib/profile2_page.inc
+++ b/contrib/profile2_page.inc
@@ -65,7 +65,10 @@ function profile2_form($form, &$form_state, $profile) {
 function profile2_form_submit($form, &$form_state) {
   // The profile is being saved by profile2_form_submit_handler().
   drupal_set_message(t('The changes have been saved.'));
-  $form_state['redirect'] = $form_state['profile2']->path();
+  // Probably user have no access to view profile, then we stay on edit.
+  $form_state['redirect'] = profile2_access('view', $form_state['profile2'])
+    ? $form_state['profile2']->path()
+    : $form_state['profile2']->path() . '/edit';
 }
 
 /**
