diff --git a/includes/bootstrap.inc b/includes/bootstrap.inc
index 490ac45..5235941 100644
--- a/includes/bootstrap.inc
+++ b/includes/bootstrap.inc
@@ -1500,7 +1500,7 @@ function drupal_page_header() {
 
   $default_headers = array(
     'Expires' => 'Sun, 19 Nov 1978 05:00:00 GMT',
-    'Cache-Control' => 'no-cache, must-revalidate',
+    'Cache-Control' => 'no-cache, no-store, must-revalidate',
     // Prevent browsers from sniffing a response and picking a MIME type
     // different from the declared content-type, since that can lead to
     // XSS and other vulnerabilities.
