diff --git a/js/dependent.js b/js/dependent.js
index e9e2447..550c90e 100644
--- a/js/dependent.js
+++ b/js/dependent.js
@@ -97,7 +97,13 @@
           else {
             switch ($(trigger).attr('type')) {
               case 'checkbox':
-                var val = $(trigger).attr('checked') ? true : false;
+                // **This check determines if using a jQuery version 1.7 or newer which requires the use of the prop function instead of the attr function when not called on an attribute
+                if ($().prop) {
+                  var val = $(trigger).prop('checked') ? true : false;
+                }
+                else {
+                  var val = $(trigger).attr('checked') ? true : false;
+                }
 
                 if (val) {
                   $(trigger).siblings('label').removeClass('hidden-options').addClass('expanded-options');
@@ -166,16 +172,31 @@
               if (Drupal.settings.CTools.dependent[id].type == 'disable') {
                 if (Drupal.settings.CTools.dependent[id].num <= len) {
                   // Show if the element if criteria is matched
-                  object.attr('disabled', false);
-                  object.addClass('dependent-options');
-                  object.children().attr('disabled', false);
+                  // **This check determines if using a jQuery version 1.7 or newer which requires the use of the prop function instead of the attr function when not called on an attribute
+                  if (typeof $().prop == 'function') {
+                    object.prop('disabled', false);
+                    object.addClass('dependent-options');
+                    object.children().prop('disabled', false);
+                  }
+                  else {
+                    object.attr('disabled', false);
+                    object.addClass('dependent-options');
+                    object.children().attr('disabled', false);
+                  }
                 }
                 else {
                   // Otherwise hide. Use css rather than hide() because hide()
                   // does not work if the item is already hidden, for example,
                   // in a collapsed fieldset.
-                  object.attr('disabled', true);
-                  object.children().attr('disabled', true);
+                  // **This check determines if using a jQuery version 1.7 or newer which requires the use of the prop function instead of the attr function when not called on an attribute
+                  if (typeof $().prop == 'function') {
+                    object.prop('disabled', true);
+                    object.children().prop('disabled', true);
+                  }
+                  else {
+                    object.attr('disabled', true);
+                    object.children().attr('disabled', true);
+                  }
                 }
               }
               else {
