diff --git a/modules/file/file.module b/modules/file/file.module
index eeb0149..b73d904 100644
--- a/modules/file/file.module
+++ b/modules/file/file.module
@@ -275,9 +275,9 @@ function file_ajax_upload() {
   if (isset($form['#file_upload_delta']) && $current_file_count < $form['#file_upload_delta']) {
     $form[$current_file_count]['#attributes']['class'][] = 'ajax-new-content';
   }
-  // Otherwise just add the new content class on a placeholder.
+  // Otherwise just add the class to the container.
   else {
-    $form['#suffix'] .= '<span class="ajax-new-content"></span>';
+    $form['#attributes']['class'][] = 'ajax-new-content';
   }
 
   $form['#prefix'] .= theme('status_messages');
