diff --git a/features_banish.module b/features_banish.module
index 3a65817..524c737 100644
--- a/features_banish.module
+++ b/features_banish.module
@@ -44,10 +44,29 @@ function features_banish_features_export_options_alter(&$options, $component) {
 
 /**
  * Helper function to unset a list of items from an array.
+ *
+ * @param array $items
+ * @param array $root_array
  */
 function features_banish_unset_items($items, &$root_array) {
-  foreach($items as $item) {
-    unset($root_array[$item]);
+  foreach ($items as $item) {
+    // Exclude all elements if '*' directly specified.
+    if ($item === '*') {
+      $root_array = array();
+    }
+    // And exclude elements starting with mask.
+    elseif (strpos($item, '*') !== FALSE) {
+      $item = str_replace('*', '', $item);
+      foreach ($root_array as $key => $value) {
+        if (strpos($key, $item) === 0) {
+          unset($root_array[$key]);
+        }
+      }
+    }
+    // If there is no mask, exclude by name directly.
+    elseif (isset($root_array[$item])) {
+      unset($root_array[$item]);
+    }
   }
 }
 
