diff --git a/single_user_role.module b/single_user_role.module
index a00617e..b890031 100755
--- a/single_user_role.module
+++ b/single_user_role.module
@@ -38,6 +38,31 @@ function single_user_role_form_alter(&$form, &$form_state, $form_id) {
       }
     }
     $form['account'][$role_field_name][2]['#access'] = FALSE;
+
+    // Make our validation first one to change structure of roles field back.
+    array_unshift($form['#validate'], 'single_user_role_form_validate');
+  }
+}
+
+/**
+ * Validate handler for user profile forms.
+ *
+ * @param $form
+ * @param $form_state
+ */
+function single_user_role_form_validate($form, &$form_state) {
+  $role_field_name = 'roles';
+  if (module_exists('role_delegation')) {
+    if (!user_access('administer permissions')) {
+      $role_field_name = 'roles_change';
+    }
+  }
+  if (!is_array($form_state['values'][$role_field_name])) {
+    // Set authenticated user role with single role selected.
+    $form_state['values'][$role_field_name] = array(
+      $form_state['values'][$role_field_name] => $form_state['values'][$role_field_name],
+      DRUPAL_AUTHENTICATED_RID => TRUE,
+    );
   }
 }
 
@@ -72,19 +97,6 @@ function single_user_role_permission() {
 }
 
 /**
- * Implements hook_user_presave().
- */
-function single_user_role_user_presave(&$edit, $account, $category) {
-  if (!is_array($edit['roles'])) {
-    // Set authenticated user role with single role selected.
-    $edit['roles'] = array(
-      $edit['roles'] => $edit['roles'],
-      2 => '1',
-    );
-  }
-}
-
-/**
  * Implements hook_help().
  */
 function single_user_role_help($path, $arg) {
