diff --git a/includes/ajax.inc b/includes/ajax.inc
index bef3c05..95b7dee 100644
--- a/includes/ajax.inc
+++ b/includes/ajax.inc
@@ -25,7 +25,9 @@ function views_ajax() {
 
     $commands = array();
 
-    // Remove all of this stuff from $_GET so it doesn't end up in pagers and tablesort URLs.
+    // Remove all of this stuff from $_GET so it doesn't end up in pagers and tablesort URLs;
+    // Do not modify $_POST itself but make a new "clean" copy to merge it with $_GET later.
+    $cleaned_POST = $_POST;
     foreach (array('view_name', 'view_display_id', 'view_args', 'view_path', 'view_dom_id', 'pager_element', 'view_base_path', 'ajax_html_ids', 'ajax_page_state') as $key) {
       if (isset($_GET[$key])) {
         unset($_GET[$key]);
@@ -33,8 +35,8 @@ function views_ajax() {
       if (isset($_REQUEST[$key])) {
         unset($_REQUEST[$key]);
       }
-      if (isset($_POST[$key])) {
-        unset($_POST[$key]);
+      if (isset($cleaned_POST[$key])) {
+        unset($cleaned_POST[$key]);
       }
     }
 
@@ -48,7 +50,7 @@ function views_ajax() {
 
       // Add all $_POST data, because AJAX is always a post and many things,
       // such as tablesorts, exposed filters and paging assume $_GET.
-      $_GET = $_POST + drupal_get_query_parameters($_GET, array('page'));
+      $_GET = $cleaned_POST + drupal_get_query_parameters($_GET, array('page'));
 
       // Overwrite the destination.
       // @see drupal_get_destination()
