diff --git a/saveguard.module b/saveguard.module
index b17bbb3..b6b883d 100644
--- a/saveguard.module
+++ b/saveguard.module
@@ -26,18 +26,71 @@ function saveguard_menu() {
 /**
  * Define a settings form.
  */
-function saveguard_admin_settings() {
-  $form = array();
+function saveguard_admin_settings($form, $form_state) {
 
   $form['saveguard_message'] = array(
     '#type' => 'textfield',
     '#title' => t('Popup Message'),
     '#default_value' => variable_get('saveguard_message', NULL),
   );
-
+  $saveguard_forms_all = variable_get('saveguard_forms_all', 0);
+  $form['saveguard_forms_all'] = array(
+    '#type' => 'radios',
+    '#title' => t('Where it should apply'),
+    '#description' => t('Specify on which forms you want to apply the <strong>SaveGuard</strong>.'),
+    '#options' => array(
+      0 => t('All forms'), 
+      1 => t("Specify <strong>form_id</strong>'s"),
+      2 => t("All forms except these <strong>form_id</strong>'s"),
+    ),
+    '#default_value' => $saveguard_forms_all,
+    '#ajax' => array(
+      'callback' => 'saveguard_forms_all_choice',
+      'wrapper' => 'forms-all-lists',
+      'method' => 'replace',
+      'effect' => 'fade',
+    ),
+  );
+  $saveguard_forms_value = variable_get('saveguard_forms', '');
+  $saveguard_except_forms_value = variable_get('saveguard_except_forms', '');
+  $form['saveguard_forms_container'] = array(
+    '#type' => 'container',
+    '#prefix' => '<div id="forms-all-lists">',
+    '#suffix' => '</div>',
+  );
+  $form['saveguard_forms_container']['saveguard_forms'] = array(
+    '#type' => 'hidden',
+    '#default_value' => $saveguard_forms_value,
+  );
+  $form['saveguard_forms_container']['saveguard_except_forms'] = array(
+    '#type' => 'hidden',
+    '#default_value' => $saveguard_except_forms_value,
+  );
+  if ((isset($form_state['values']['saveguard_forms_all']) && $form_state['values']['saveguard_forms_all'] == 1) || (!isset($form_state['values']['saveguard_forms_all']) && $saveguard_forms_all == 1)) {
+    $form['saveguard_forms_container']['saveguard_forms'] = array(
+      '#type' => 'textarea',
+      '#title' => t("List of form_id's"),
+      '#description' => t("Specify a list of all form_id's where you want SaveGuard to apply."),
+      '#default_value' => $saveguard_forms_value,
+    );
+  }
+  if ((isset($form_state['values']['saveguard_forms_all']) && $form_state['values']['saveguard_forms_all'] == 2) || (!isset($form_state['values']['saveguard_forms_all']) && $saveguard_forms_all == 2)) {
+    $form['saveguard_forms_container']['saveguard_except_forms'] = array(
+      '#type' => 'textarea',
+      '#title' => t("List of form_id's exceptions"),
+      '#description' => t("Specify a list of exceptions (form_ids) where you want SaveGuard not to apply."),
+      '#default_value' => $saveguard_except_forms_value,
+    );
+  }
   return system_settings_form($form);
 }
 
+/**
+ * AJAX Callback for the radios choice.
+ */
+function saveguard_forms_all_choice($form, $form_state) {
+  return $form['saveguard_forms_container'];
+}
 
 /**
  * Implements hook_form_alter().
@@ -50,6 +103,45 @@ function saveguard_admin_settings() {
  *    array with form structure
  */
 function saveguard_form_alter(&$form, &$form_state, $form_id) {
+  static $saveguard_form_ids_string;
+  static $saveguard_except_forms_ids_string;
+
+  $form_opt_type = variable_get('saveguard_forms_all', 0);
+  if (!isset($saveguard_form_ids_string)) {
+    $saveguard_form_ids_string = variable_get('saveguard_forms', '');
+  }
+  if (!isset($saveguard_except_forms_ids_string)) {
+    $saveguard_except_forms_ids_string = variable_get('saveguard_except_forms', '');
+  }
+
+  switch ($form_opt_type) {
+    case 0:
+      // We simply apply to all forms (default).
+      _saveguard_add_form_guard();
+      break;
+
+    case 1:
+      // We apply only to specified form_id's.
+      $saveguard_form_ids = list_extract_allowed_values($saveguard_form_ids_string, 'list_number', TRUE);
+      if (count($saveguard_form_ids) > 0 && in_array($form_id, $saveguard_form_ids)) {
+        _saveguard_add_form_guard();
+      }
+
+      break;
+    case 2:
+      // We apply to all forms except the specified form_ids.
+      $saveguard_except_forms_ids = list_extract_allowed_values($saveguard_except_forms_ids_string, 'list_number', TRUE);
+      if (!in_array($form_id, $saveguard_except_forms_ids)) {
+        _saveguard_add_form_guard();
+      }
+      break;
+  }
+}
+
+/**
+ * Private function that adds the saveguard itself.
+ */
+function _saveguard_add_form_guard() {
   drupal_add_js(drupal_get_path('module', 'saveguard') . '/saveguard.js');
   static $done;
   if (!$done) {
