diff --git a/includes/form.inc b/includes/form.inc
index 130775f..47219d3 100644
--- a/includes/form.inc
+++ b/includes/form.inc
@@ -3876,6 +3876,22 @@ function form_process_vertical_tabs($element, &$form_state) {
  */
 function theme_vertical_tabs($variables) {
   $element = $variables['element'];
+
+  // If active tab element is the only element with rendered content,
+  // render nothing for vertical tabs.
+  $active_tab_field = implode('__', $element['#parents'])  . '__active_tab';
+  $has_children = FALSE;
+  foreach (element_children($element) as $child_key) {
+    if (!empty($element[$child_key]['#children']) && $child_key !== $active_tab_field) {
+      $has_children = TRUE;
+      break;
+    }
+  }
+
+  if (!$has_children) {
+    return '';
+  }
+
   // Add required JavaScript and Stylesheet.
   drupal_add_library('system', 'drupal.vertical-tabs');
 
