diff --git a/includes/form.inc b/includes/form.inc
index 130775f..c5e74b2 100644
--- a/includes/form.inc
+++ b/includes/form.inc
@@ -1463,7 +1463,15 @@ function _form_validate(&$elements, &$form_state, $form_id = NULL) {
     // #value data.
     elseif (isset($elements['#element_validate'])) {
       foreach ($elements['#element_validate'] as $function) {
-        $function($elements, $form_state, $form_state['complete form']);
+        // Run validation only if an element is accessible.
+        if (!isset($elements['#access']) || $elements['#access']) {
+          $function($elements, $form_state, $form_state['complete form']);
+        }
+        // Or for some exceptions where validation modify element.
+        elseif ($function == 'options_field_widget_validate') {
+          $items = _options_form_to_storage($elements);
+          form_set_value($elements, $items, $form_state);
+        }
       }
     }
     $elements['#validated'] = TRUE;
