diff --git a/includes/form.inc b/includes/form.inc
index 130775f..60b6403 100644
--- a/includes/form.inc
+++ b/includes/form.inc
@@ -1461,7 +1461,7 @@ function _form_validate(&$elements, &$form_state, $form_id = NULL) {
     }
     // Call any element-specific validators. These must act on the element
     // #value data.
-    elseif (isset($elements['#element_validate'])) {
+    elseif ((!isset($elements['#access']) || $elements['#access']) && isset($elements['#element_validate'])) {
       foreach ($elements['#element_validate'] as $function) {
         $function($elements, $form_state, $form_state['complete form']);
       }
