diff --git a/ife.settings.inc b/ife.settings.inc
index cee3cbc..cbbac92 100644
--- a/ife.settings.inc
+++ b/ife.settings.inc
@@ -99,8 +99,13 @@ function ife_settings_form($form, $form_state) {
     '#options' => ife_message_display_options(),
     '#default_value' => IFE_MESSAGE_DEFAULT,
   );
+  // Submit button.
+  $form['submit'] = array(
+  '#type' => 'submit',
+  '#value' => t('Save configuration'),
+  );
 
-  return system_settings_form($form);
+  return $form;
 }
 
 /**
@@ -138,7 +143,13 @@ function ife_settings_form_validate($form, &$form_state) {
  */
 function ife_settings_form_submit($form, &$form_state) {
   $values = $form_state['values'];
-  
+
+  // Set general options.
+  variable_set('ife_show_form_ids', $values['ife_show_form_ids']);
+  variable_set('ife_display', $values['ife_display']);
+  variable_set('ife_position_inline_message', $values['ife_position_inline_message']);
+  variable_set('ife_general_message', $values['ife_general_message']);
+
   // Write form_ids to the database.
   $form_ids = $values['form_ids'];
   array_pop($form_ids);
